src/components/SearchResult/SearchResult.js

```diff
@@ -1,55 +1,168 @@
+import CellMeasurer, { CellMeasurerCache } from "react-virtualized/dist/commonjs/CellMeasurer";
+import ListSeparator from "components/ListSeparator";
 
-class SearchResult extends React.PureComponent {
-  static propTypes = {
-    index: PropTypes.number.isRequired,
-    result: PropTypes.object.isRequired,
-    activeResultIndex: PropTypes.number.isRequired,
-    onClickResult: PropTypes.func.isRequired,
-  };
+const SearchResultListSeparatorPropTypes = {
+  currentResultIndex: PropTypes.number.isRequired,
+  searchResults: PropTypes.arrayOf(PropTypes.object).isRequired,
+  translate: PropTypes.func.isRequired,
+  pageLabels: PropTypes.arrayOf(PropTypes.any).isRequired,
+};
 
-  onClick = () => {
-    const { onClickResult, index, result } = this.props;
+function SearchResultListSeparator(props) {
+  const { currentResultIndex, searchResults, translate, pageLabels } = props;
 
-    onClickResult(index, result);
-  }
+  const previousIndex = currentResultIndex === 0 ? currentResultIndex : currentResultIndex - 1;
+  const currentListItem = searchResults[currentResultIndex];
+  const previousListItem = searchResults[previousIndex];
 
-  renderContent = () => {
-    const { ambientStr, resultStrStart, resultStrEnd, resultStr } = this.props.result;
-    const textBeforeSearchValue = ambientStr.slice(0, resultStrStart);
-    const searchValue = ambientStr === '' ? resultStr : ambientStr.slice(resultStrStart, resultStrEnd);
-    const textAfterSearchValue = ambientStr.slice(resultStrEnd);
+  const isFirstListItem = previousListItem === currentListItem;
+  const isInDifferentPage = previousListItem.pageNum !== currentListItem.pageNum;
 
+  if (isFirstListItem || isInDifferentPage) {
+    const listSeparatorText = `${translate('option.shared.page')} ${pageLabels[currentListItem.pageNum - 1]}`;
     return (
-      <React.Fragment>
-        {textBeforeSearchValue}
-        <span className="search-value">
-          {searchValue}
-        </span>
-        {textAfterSearchValue}
-      </React.Fragment>
+      <ListSeparator>{listSeparatorText}</ListSeparator>
     );
   }
+  return null;
+}
+
+SearchResultListSeparator.propTypes = SearchResultListSeparatorPropTypes;
+
+const SearchResultListItemPropTypes = {
+  result: PropTypes.object.isRequired,
+  currentResultIndex: PropTypes.number.isRequired,
+  activeResultIndex: PropTypes.number.isRequired,
+  onSearchResultClick: PropTypes.func,
+};
+
+function SearchResultListItem(props) {
+  const { result, currentResultIndex, activeResultIndex, onSearchResultClick } = props;
+  const { ambientStr, resultStrStart, resultStrEnd, resultStr } = result;
+  const textBeforeSearchValue = ambientStr.slice(0, resultStrStart);
+  const searchValue = ambientStr === '' ? resultStr : ambientStr.slice(resultStrStart, resultStrEnd);
+  const textAfterSearchValue = ambientStr.slice(resultStrEnd);
+  return (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+    <div
```

Comment:
Hmm I'm not very sure if we just want to ignore linting here...

@liamross when you have time can you take a look here and see if it's ok. Thanks.

Reply:
Original code was giving 2 different linting errors. First one was easy fix, but I wasn't sure if we want to add key events to every search result item, so I disabled for now. Happy to add key events there as well if we have defined way how we are usually doing these

Reply:
Yeah generally there are two moves: convert to a button then use my `@include button-reset;` sass style to remove button-specific styling (or at least a bulk of it), or add onKeyDown listeners for <kbd>Space</kbd> and <kbd>Enter</kbd>

Reply:
Converted this to use `<button>`